lib/npm.js

```diff
@@ -137,7 +137,19 @@ const npm = module.exports = new class extends EventEmitter {
 
     const workspacesEnabled = this.config.get('workspaces')
     const workspacesFilters = this.config.get('workspace')
+    if (workspacesEnabled === false && workspacesFilters.length > 0)
+      return cb(new Error('Can not use --no-workspaces and --workspace at the same time'))
+
     const filterByWorkspaces = workspacesEnabled || workspacesFilters.length > 0
+    // normally this would go in the constructor, but our tests don't
+    // actually use a real npm object so this.npm.config isn't always
+    // populated.  this is the compromise until we can make that a reality
+    // and then move this into the constructor.
+    impl.workspaces = this.config.get('npm')
+    impl.workspacePaths = null
+    // normally this would be evaluated in base-command#setWorkspaces, see
+    // above for explanation
+    impl.includeWorkspaceRoot = this.config.get('include-workspace-root')
```

Comment:
```suggestion
    impl.includeWorkspaceRoot = this.config.get('include-workspace-root') || workspacesFilters.includes(pkg.name)
```
In that case would this work? Re: https://github.com/npm/cli/pull/3587#discussion_r691697815

Reply:
No, your root package name won't parse as a valid workspace, for the same reasons mentioned.

```sh
~/D/n/cli (release-next|✔) $ npm repo --workspace npm --no-browser
npm ERR! No workspaces found:
npm ERR!   --workspace=npm

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/wraithgar/.npm/_logs/2021-08-19T01_58_09_831Z-debug.log
~/D/n/cli (release-next|✔) $ npm repo --workspace docs --no-browser
docs repo available at the following URL:
  https://github.com/npm/cli
```

Reply:
```Shell
$ npm repo --workspace npm --include-workspace-root --no-browser
npm repo available at the following URL:
  https://github.com/npm/cli
```
`--workspace npm` already works if you add `--include-workspace-root` -- wouldn't the suggested change do just the same thing? (`includeWorkspaceRoot ||= workspacesFilters.includes(pkg.name)`)